src/binary.ts

```diff
@@ -292,7 +302,13 @@ export class Binary extends BSONValue {
   }
 
   inspect(): string {
-    return `new Binary(Buffer.from("${ByteUtils.toHex(this.buffer)}", "hex"), ${this.sub_type})`;
+    if (this.position === 0) {
```

Comment:
We had a bug here where the default `new Binary()` printed:

```ts
Binary(Buffer.from("00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "hex"), 0)
```

Which isn't the same as what would end up in BSON, b/c position determines how many bytes are serialized. The above constructs a Binary where position is set to 256, instead of set to zero.